### PR TITLE
Ci/one snap per arch

### PR DESCRIPTION
**Pull request by Guillaumebeuzeboc**
One folder and `snapcraft.yaml` is generated for each distro/variant/architecture.

The push was tested [here](https://github.com/ubuntu-robotics/ros-content-sharing-snaps/actions/runs/6888443288).

All the necessary environment variable and secrets are already set in the repo.

### CODE REVIEW

File: .github/workflows/monthly-builds.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT7_CREDS }}
+      # snapcraft remote-build will create a repository with the name decided by the --build-id arg
+      # URL to this repo echoed below to help debug builds (does not change if the workflow is re-run)
+      SNAPCRAFT_BUILDER_ID: ${{ github.run_id }}
+    steps:
+      - name: Download snapcrafts folders
+        uses: actions/download-artifact@v3
+        with:
+          name: snaps
+      - name: print matrix
```
Comment (giusebar) [inline .github/workflows/monthly-builds.yaml:59]: should we remove this?

File: .github/workflows/monthly-builds.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        name: ${{ matrix.snap_names }}
+        path: .
+    - name: Get snap filename
+      run: echo "SNAPFILE=$(ls *.snap)" >> $GITHUB_ENV
+    - uses: snapcore/action-publish@v1
+      env:
+        SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+      with:
+        snap: ${{ env.SNAPFILE }}
+        release: edge
```
Comment (giusebar) [inline .github/workflows/monthly-builds.yaml:118]: so here we are releasing on edge and then manually moving them to stable?

Comment (Guillaumebeuzeboc) [reply .github/workflows/monthly-builds.yaml:118]: I guess this is to discuss.

File: .github/workflows/monthly-builds.yaml (hunk 1 above)
Comment (Guillaumebeuzeboc) [reply .github/workflows/monthly-builds.yaml:59]: I will remove this.

**Review by giusebar — APPROVED** [review]

File: .github/workflows/monthly-builds.yaml (hunk 2 above)
Comment (Guillaumebeuzeboc) [reply .github/workflows/monthly-builds.yaml:115]: After discussion:
- we push on edge
- we must add a workflow to release from edge to stable

**Review by artivis — APPROVED** [review]

**Pull request merged**